### PR TITLE
Medulloblastoma SHH TP53 subtypes

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?
Identifying the TP53 mutated Medulloblastoma SHH samples 

#### What was your approach?
1) Read in SNV consensus and metadata
2) Filtered to Medulloblastoma
3) Tried to filter to SHH samples based on `molecular_subtype` but that lead to 0 samples for SNV. Is the `molecular_subtype` column complete? - after talking with @jaclyn-taroni , I've realized that the subtyping labels are not necessarily propagated to other `Biospecimen` ids from the `sample_ids`. I will do this temporarily in this notebook and file a issue for a request for these SHH labels to be propagated in the original metadata. 
4) Tallied the number of TP53 mutations for all medulloblastoma samples. 

#### What GitHub issue does your pull request address?
#247

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

**Questions about this analysis as well as molecular subtyping analyses in general**:
- How do we want this output data saved? Do we need a list of TP53 mutated vs non-mutated? What is ideal for it to be added to a future data release?
- The issue, #247 mentions that I will need SV, CNV, fusion data. Is this accurate? If so, how should these data be used for this identification. Here I have only used SNV consensus calls since that seemed most straightforward and informative to me. 

#### What types of results are included (e.g., table, figure)?

You can see the rendered notebook here: https://cansavvy.github.io/openpbta-notebook-concept/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.nb.html

#### Reproducibility Checklist
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile. No new dependencies needed. 
- [x] This analysis has been added to continuous integration.
- [x] This analysis is recorded in the table in `analyses/README.md`.

### CODE REVIEW

File: analyses/README.md [hunk 1]
```diff
@@ -11,26 +11,28 @@ Note that _nearly all_ modules use the harmonized clinical data file (`pbta-hist
...
 | [`molecular-subtyping-ATRT`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-ATRT) | `analyses/ssgsea-hallmark/results/GeneSetExpressionMatrix.RDS`, `pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds`, `analyses/focal-cn-file-preparation/results/controlfreec_annotated_cn_autosomes.tsv.gz`, `pbta-snv-consensus-mutation-tmb.tsv` | *In progress*; summarizing data into tabular format in order to molecularly subtype ATRT samples [#244](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/244) | N/A
-| [`mutational-signatures`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/mutational-signatures) | `pbta-snv-consensus-mutation.maf.tsv.gz` | Performs COSMIC and Alexandrov et al. mutational signature analysis using the consensus SNV data | N/A 
+| [`molecular-subtyping-SHH-tp53`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-SHH-tp53) | `pbta-histologies` and `pbta-snv-consensus-mutation-tmb.tsv` | Identify the medulloblastoma samples that have TP53 mutations | N/A
+
```
Comment (jaclyn-taroni) [inline analyses/README.md]:
```suggestion
```

File: analyses/README.md [hunk 2]
```diff
@@ -11,26 +11,28 @@ Note that _nearly all_ modules use the harmonized clinical data file (`pbta-hist
...
 | [`molecular-subtyping-ATRT`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-ATRT) | `analyses/ssgsea-hallmark/results/GeneSetExpressionMatrix.RDS`, `pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds`, `analyses/focal-cn-file-preparation/results/controlfreec_annotated_cn_autosomes.tsv.gz`, `pbta-snv-consensus-mutation-tmb.tsv` | *In progress*; summarizing data into tabular format in order to molecularly subtype ATRT samples [#244](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/244) | N/A
-| [`mutational-signatures`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/mutational-signatures) | `pbta-snv-consensus-mutation.maf.tsv.gz` | Performs COSMIC and Alexandrov et al. mutational signature analysis using the consensus SNV data | N/A 
+| [`molecular-subtyping-SHH-tp53`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-SHH-tp53) | `pbta-histologies` and `pbta-snv-consensus-mutation-tmb.tsv` | Identify the medulloblastoma samples that have TP53 mutations | N/A
```
Comment (jaclyn-taroni) [inline analyses/README.md]:
```suggestion
| [`molecular-subtyping-SHH-tp53`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-SHH-tp53) | `pbta-histologies` and `pbta-snv-consensus-mutation-tmb.tsv` | Identify the SHH-classified medulloblastoma samples that have TP53 mutations | N/A
```

File: analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+# Make into list form so it can be used in dplyr::recode
+key <- key_df$molecular_subtype
+names(key) <- key_df$sample_id
+```
+
+Recode and filter to all `sample_id`s that are `SHH`.
+
+```{r}
+metadata <- metadata %>% 
+  dplyr::mutate(molecular_subtype = dplyr::recode(sample_id, !!!key, .default = "NA")) %>%   dplyr::filter(molecular_subtype == "SHH")
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd]:
```suggestion
  dplyr::mutate(molecular_subtype = dplyr::recode(sample_id, !!!key, .default = "NA")) %>%
  dplyr::filter(molecular_subtype == "SHH")
```

**Comment by jaclyn-taroni** [comment]
@cansavvy I think what we want in essence is the following table, where the samples in this table are only SHH MB samples:

| Kids_First_Participant_ID | sample_id | Kids_First_Biospecimen_ID(s) | _TP53_ mutation |
|----|----|----|----|

And the _TP53_ column is binary. Note that you'll probably need to collapse the biospecimen IDs through a combination of `mutate` and `paste`...

File: analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+
+# Print out this table
+tp53_status_table
+```
+
+Save table to a TSV file. 
+
+```{r}
+readr::write_tsv(tp53_mut, file.path(results_dir, "tp53-shh-samples-status.tsv"))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd]: Should this be
```suggestion
readr::write_tsv(tp53_status_table, file.path(results_dir, "tp53-shh-samples-status.tsv"))
```

File: analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+names(key) <- shh_metadata$Kids_First_Biospecimen_ID
+
+# List of tp53 postive samples
+tp53_mut_samples <- dplyr::recode(tp53_mut_biospecimens, !!!key)
+```
+
+Summarize TP53 status to a table with other sample identifiers.
+
+```{r}
+tp53_status_table <- shh_metadata %>%
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd]: It does look like there are cases where a `sample_id` only has an RNA-seq sample and no WGS sample (e.g., `7316-1884`). I would say that that is _distinct_ from the case where you have a WGS sample and no mutation and should probably be a `NA` in `TP53_mutation`.

Comment (cansavvy) [reply analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd]: Okay. I will add in a fix for this kind of case.

**Pull request merged**